lib/app.rb

```diff
@@ -0,0 +1,90 @@
+require "tic_tac_toe"
+
+require_relative 'datastore'
+require_relative 'player'
+require_relative 'output'
+require_relative 'validate'
+
+require 'sinatra'
+require "json"
+
+
+class App < Sinatra::Base
+
+    def initialize(app=nil, datastore={}, turn=TicTacToe::Turn.new, output = Output.new, game=TicTacToe::Game.new, validate=Validate.new)
+        super(app)
+        @datastore = datastore
+        @turn = turn
+        @game = game
+        @output = output
+        @validate = validate
+    end
+
+    before do
+        content_type :json
+    end
+
+    get '/games' do
+        {"games":@datastore.load_all()}.to_json
+    end
+
+    get '/' do
+        {'message':@output.welcome}.to_json
+    end
+
+    get '/available_moves/:game_id' do
+        game_id = params['game_id'].to_i
+        if @validate.validate_available_moves(game_id, @datastore)
```

Comment:
Based on the implementation of `#validate_available_moves`, it looks like it is actually checking if a game has started rather than if there are available moves. Perhaps a better method name?